README.md

```diff
@@ -8,34 +8,318 @@ or
+fetch_store = TeePublic::Api.send(store_url).deep_symbolize_keys!
+affiliate_network_id = fetch_store.dig(:headers, :'x-affiliatenetwork-id')
+puts affiliate_network_id
+```
+
+## Step 4 -  Add Routes to routes.rb file
+Application will need at-least below three routes. One for teepublic store, one for teepublic design and one for cart page.
+This is how our demo application creates these routes
+```
+get '/stores/:id', to: 'application#store'
+get '/stores/:id/designs/:product_type/:design_id', to: 'application#design'
+get '/cart', to: 'application#cart'
+```
+Decide which URL you plan to dedicate to show teepublic pages and add them to your `routes.rb`
+
+
+## Step 5 - Prepare Controller
+According to Step 5, add actions in controller to handle newly added routes.
+
+**Please refactor according to the application style, we added all logic into one function to make it easier to understand**
+
+#### Store action
+```
+def store
+  @page ||= params['page'].try(:to_i) || 1
+  store_url = "stores/#{params[:id]}?page=#{@page}"
+  store_url += "&album_id=#{params['album_id']}" if params['album_id'].present?
+  store_url += "&product_type=#{params['product_type']}" if params['product_type'].present?
+
+  fetch_store = TeePublic::Api.send(store_url).deep_symbolize_keys!
+  @store = fetch_store[:body]
+end
+```
+
+#### Design Action
+```
+def design
+  @page ||= params['page'].try(:to_i) || 1
+  store_url = "stores/#{params[:id]}?page=#{@page}"
+  store_url += "&album_id=#{params['album_id']}" if params['album_id'].present?
+  store_url += "&product_type=#{params['product_type']}" if params['product_type'].present?
+
+  fetch_store = TeePublic::Api.send(store_url).deep_symbolize_keys!
+  @store = fetch_store[:body]
+
+  fetch_design = TeePublic::Api.designs(params['design_id']).deep_symbolize_keys!
+  @design = fetch_design[:body]
+
+
+  skus_url = "designs/#{params['design_id']}/products/#{params['product_type']}"
+  fetch_skus = TeePublic::Api.send(skus_url).deep_symbolize_keys!
+  @skus = fetch_skus[:body]
+
+  # Set at the end since this requires a response.
+  @affiliate_network_id = fetch_store.dig(:headers, :'x-affiliatenetwork-id')
+end
 ```
-import { DesignCollection } from 'teepublic-react'
 
-class MyStore extends React.Component {
+#### Cart action
+```
+def cart
+end
+```
+
+## Step 7 - Creating views
+Create 3 corresponding views to actions added in step 6
+
+#### Store view
+Add store.html.erb
+```
+<%= react_component('TeePublicStore', store: @store, page: @page, albumId: params[:album_id], productTypeName: params[:product_type]) %>
+```
+
+#### Design view
+Add design.html.erb
+```
+<%= react_component('TeePublicDesign', store: @store, design: @design, skuData: @skus, affiliateNetworkId: @affiliate_network_id) %>
+```
+
+#### Cart view
+Add cart.html.erb
+```
+<%= react_component('TeePublicCart') %>
+```
+
+## Step 8 - Creating TeePublic React Components
+Let's install the latest version of teepublic-react package
```

Comment:
Watch filler conversational words like "Let's"

Reply:
Ok, I will take an another pass on it